textx/scoping/providers.py

```diff
@@ -462,9 +468,13 @@ def register_models(self, filename_pattern):
 
     def _load_referenced_models(self, model, encoding):
         for filename_pattern in self.filename_pattern_list:
+            if not isabs(filename_pattern) and \
```

Comment:
`filename_pattern = join(model._tx_model_params.get('project_root', os.getcwd()), filename_pattern)` should work even if `filename_pattern` is absolute path.

Reply:
But if an absolute path is passed I think it should be left as it is... (my opinion). @igordejanovic what do you think?

Reply:
But if an absolute path is passed I think it should be left as it is... (my opinion). @igordejanovic what do you think?

Reply:
But if an absolute path is passed I think it should be left as it is... (my opinion). @igordejanovic what do you think?

Reply:
From [docs](https://docs.python.org/3/library/os.path.html?highlight=os%20path%20join#os.path.join): 
> ...
If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.
...

`join('/a/b/c', '/d/e/f')` returns `/d/e/f`.

Reply:
OK. You say, I do not need the 'if', because `join` already does the job... Good point. But the behavior will remain as it is...